vcd/datasource_vcd_network_test.go

```diff
@@ -0,0 +1,367 @@
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_routed"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no routed network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"VDC":         testConfig.VCD.Vdc,
+		"EdgeGateway": data.parent,
+		"NetworkName": data.network.Name,
+		"FuncName":    "TestNetworkRoutedDS",
+		"Tags":        "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { preRunChecks(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckOutput("network_org", testConfig.VCD.Org),
+					resource.TestCheckOutput("network_vdc", testConfig.VCD.Vdc),
+					resource.TestCheckOutput("network_name", data.network.Name),
+					resource.TestCheckOutput("network_edge", data.parent),
+					resource.TestCheckOutput("default_gateway", testConfig.Networking.ExternalNetwork),
+				),
+			},
+		},
+	})
+}
+
+func TestAccVcdNetworkIsolatedDS(t *testing.T) {
+	// This test requires access to the vCD before filling templates
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	err := getAvailableNetworks()
+
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		t.Skip("error getting available networks")
+		return
+	}
+	if len(availableNetworks) == 0 {
+		t.Skip("No networks found - data source test skipped")
+		return
+	}
+
+	networkType := "vcd_network_isolated"
+	data, ok := availableNetworks[networkType]
+	if !ok {
+		t.Skip("no isolated network found ")
+		return
+	}
+
+	template := networkTemplates[networkType]
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"VDC":         testConfig.VCD.Vdc,
+		"NetworkName": data.network.Name,
+		"FuncName":    "TestNetworkIsolatedDS",
+		"Tags":        "network",
+	}
+	configText := templateFill(template, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { preRunChecks(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckOutput("network_org", testConfig.VCD.Org),
+					resource.TestCheckOutput("network_vdc", testConfig.VCD.Vdc),
+					resource.TestCheckOutput("network_name", data.network.Name),
+					resource.TestCheckOutput("network_gateway", data.network.Configuration.IPScopes.IPScope[0].Gateway),
+					resource.TestCheckOutput("network_netmask", data.network.Configuration.IPScopes.IPScope[0].Netmask),
```

Comment:
Thinking out loud. This checks that the output variables are defined correctly, but does not validate if the variable itself was or was not set. My point - if the value by `data.network.Configuration.IPScopes.IPScope[0].Netmask` was actually empty - the test would still pass, wouldn't it? Same would apply to "network_name" and "network_gateway".
Would it make sense to do some additional checks (maybe based on regex) that the values are actually set to something we expect?

Reply:
In this particular test, we are checking the data against a live network structure (data.network), and therefore such fields cannot be empty.

Reply:
Sorry, I overlooked the variable name. I thought this was comparing `data` source field with `output` field.